cmdstanpy/utils.py

```diff
@@ -216,9 +216,13 @@ def rdump(path: str, data: Dict) -> None:
             fd.write('\n')
 
 
-def check_csv(path: str, is_optimizing: bool = False) -> Dict:
+def check_csv(
+    path: str,
+    is_optimizing: bool = False,
+    is_sampling: bool = True
+) -> Dict:
     """Capture essential config, shape from stan_csv file."""
-    meta = scan_stan_csv(path, is_optimizing=is_optimizing)
+    meta = scan_stan_csv(path, is_sampling=is_sampling)
```

Comment:
I think this might be too limited - still to be added are methods for ADVI and fixed_param - might need higher-level redesign.
for now the question is whether or not we have enough test coverage of this logic to make sure that if StanFit object is result of optimization, it behaves properly - do we?